pkg/endpoint/policy.go

```diff
@@ -825,7 +825,10 @@ func (e *Endpoint) UpdateNoTrackRules(annoCB AnnotationsResolverCB) {
 	}
 
 	updateRes := <-ch
-	regenResult := updateRes.(*EndpointRegenerationResult)
+	regenResult, ok := updateRes.(*EndpointRegenerationResult)
+	if !ok {
+		return
+	}
 	if regenResult.err != nil {
 		e.getLogger().WithError(regenResult.err).Error("EndpointNoTrackEvent event failed")
 	}
```

Comment:
I'd probably shorten this to:
```suggestion
	regenResult, ok := updateRes.(*EndpointRegenerationResult)
	if ok && regenResult.err != nil {
		e.getLogger().WithError(regenResult.err).Error("EndpointNoTrackEvent event failed")
	}
```

Similarly for the changes below.

Reply:
thanks :100: